astropy/table/operations.py

```diff
@@ -66,7 +67,8 @@ def _get_out_class(objs):
         if issubclass(obj.__class__, out_class):
             out_class = obj.__class__
 
-    if any(not issubclass(out_class, obj.__class__) for obj in objs):
+    if any(not (issubclass(out_class, obj.__class__) or has_info_class(obj, MixinInfo))
```

Comment:
actually this is also another problematic point and an ugly workaround for half the cases, as now e.g. Time and Column are stackable (just as Column and Time), but calling this function directly it ends up raising the ValueError for the first case.

Reply:
You should revert this diff, there is no reason it should care about whether the object is a mixin column.  It is just trying to find a common class that works for all classes in the list.  If you are actually trying to vstack a Time and Column (that have the same name in input tables), that should result in the `unmergeable object classes` exception.